server/index.js

```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+const app = express();
+const port = 3000;
+const router = require('./routes.js');
+
+app.use(express.static(__dirname + '/../client/dist'));
+app.use(express.json());
+app.use(express.urlencoded({extended: false}));
+
+app.get('/product', router);
```

Comment:
it seems the router is only set to catch the "get" request from client, and the path has to be '/product/.....'
I think this may be so that the path and methods can be more general. 
`app.use('/', router);`

Reply:
You are correct, fixed!